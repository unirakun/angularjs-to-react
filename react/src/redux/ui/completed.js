import { keyValue } from 'k-redux-factory'

export default keyValue({ path: 'ui', key: 'id', name: 'completed' })
