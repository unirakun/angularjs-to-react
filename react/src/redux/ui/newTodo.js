import { simpleObject } from 'k-redux-factory'

export default simpleObject({ path: 'ui', name: 'newTodo', defaultData: '' })
