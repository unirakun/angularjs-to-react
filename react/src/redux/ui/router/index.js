export * from './router'
export * from './router.selectors'
export * from './router.actions'
