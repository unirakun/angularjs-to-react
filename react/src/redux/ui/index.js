import { combineReducers } from 'redux'
import newTodo from './newTodo'
import completed from './completed'
import editing from './editing'

export default combineReducers({
  completed,
  editing,
  newTodo,
})
