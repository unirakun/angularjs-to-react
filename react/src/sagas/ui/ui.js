import { put, select } from 'redux-saga/effects'
import newTodo from 'redux/ui/newTodo'
import editing from 'redux/ui/editing'
import todos from 'redux/data/todos'

export function* setNewTodo(todo) {
  yield put(newTodo.set(todo))
}

export function* edit(todoId) {
  const todo = yield select(todos.get(todoId))
  yield put(editing.add(todo))
}

export function* cancelEdit(todoId) {
  yield put(editing.remove(todoId))
}

export function* setEditTodo(todo) {
  yield put(editing.update(todo))
}
