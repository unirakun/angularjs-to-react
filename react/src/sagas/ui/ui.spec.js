/* eslint-env jest */
import tester from 'k-redux-saga-tester'
import { setNewTodo, edit, cancelEdit, setEditTodo } from './ui'

describe('sagas/ui', () => {
  describe('setNewTodo', () => {
    const test = tester(setNewTodo)

    it('should set new todo', () => {
      expect(test('todo')()).toMatchSnapshot()
    })
  })

  describe('edit', () => {
    const test = tester(edit)

    it('should edit todo', () => {
      const mocks = {
        select: [() => ({ id: 'todo1', todo: 'todo' })],
      }
      expect(test('todo1')(mocks)).toMatchSnapshot()
    })
  })

  describe('cancelEdit', () => {
    const test = tester(cancelEdit)

    it('should edit todo', () => {
      expect(test('todo1')()).toMatchSnapshot()
    })
  })

  describe('setEditTodo', () => {
    const test = tester(setEditTodo)

    it('should edit todo', () => {
      expect(test('todo1')()).toMatchSnapshot()
    })
  })
})
