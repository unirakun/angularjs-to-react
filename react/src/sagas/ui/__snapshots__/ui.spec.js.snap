// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sagas/ui cancelEdit should edit todo 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "payload": "todo1",
          "type": "@@krf/REMOVE_EDITING",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": true,
    "value": undefined,
  },
]
`;

exports[`sagas/ui edit should edit todo 1`] = `
Array [
  Object {
    "raw": Object {
      "done": false,
      "value": Object {
        "@@redux-saga/IO": true,
        "SELECT": Object {
          "args": Array [],
          "selector": [Function],
        },
      },
    },
    "value": Object {
      "id": "todo1",
      "todo": "todo",
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "payload": Object {
            "id": "todo1",
            "todo": "todo",
          },
          "type": "@@krf/ADD_EDITING",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": true,
    "value": undefined,
  },
]
`;

exports[`sagas/ui setEditTodo should edit todo 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "payload": "todo1",
          "type": "@@krf/UPDATE_EDITING",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": true,
    "value": undefined,
  },
]
`;

exports[`sagas/ui setNewTodo should set new todo 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "payload": "todo",
          "type": "@@krf/SET_NEWTODO",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": true,
    "value": undefined,
  },
]
`;
